src/coreclr/jit/morph.cpp

```diff
@@ -12446,6 +12375,34 @@ void Compiler::fgAssertionGen(GenTree* tree)
 #endif
     };
 
+    // If this tree creates an assignment of 0 or 1 to an int local, also create a [0..1] subrange
+    // assertion for that local, in case this local is used as a bool.
+    //
+    auto AddImpliedBoolSubrangeAssertion = [=](AssertionIndex index, ASSERT_TP assertions) {
```

Comment:
Since this is a local I think it should be named in camelCase (most lambdas are named like that, although seems we have some that use this convention too).